src/api/java/AST.java

```diff
@@ -74,14 +63,10 @@ else if (getId() > other.getId())
      * 
      * @return A hash code
      **/
+    @Override
     public int hashCode()
     {
-        int r = 0;
-        try {
-            Native.getAstHash(getContext().nCtx(), getNativeObject());
-        }
-        catch (Z3Exception ex) {}
-        return r;
+        return Native.getAstHash(getContext().nCtx(), getNativeObject());
```

Comment:
Well yes, but the code above always returns zero.
Do you want to cache the hash value? Or just catch the exception?